tracer/src/Datadog.Trace.ClrProfiler.Native/calltarget_tokens.cpp

```diff
@@ -680,25 +680,25 @@ mdToken CallTargetTokens::GetCurrentTypeRef(const TypeInfo* currentType, bool& i
 {
     if (currentType->type_spec != mdTypeSpecNil)
     {
+        isValueType = currentType->valueType;
         return currentType->type_spec;
     }
+    else if (!currentType->isGeneric)
+    {
+        isValueType = currentType->valueType;
+        return currentType->id;
+    }
     else
     {
-
+        // If the type is a generic type we transverse the type.
```

Comment:
Now that I'm looking at this I can't believe it worked before 😅 Good fix!

Reply:
I had to change this function again to make it work with generic types as parent of a nested type in [bb06353](https://github.com/DataDog/dd-trace-dotnet/pull/2818/commits/bb063539f7439b173a3cf0496e6a94ee967dcda5)